scripts/syntactics/common/utils.py

```diff
@@ -0,0 +1,424 @@
+# coding: utf-8
+
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+import logging
+import math
+import os
+import sys
+import time
+
+import mxnet as mx
+import mxnet.ndarray as nd
+import numpy as np
+from mxnet.gluon import rnn
+from mxnet.gluon.contrib.rnn import VariationalDropoutCell
+
+from .data import ParserVocabulary
+from .tarjan import Tarjan
+
+
+class Progbar(object):
```

Comment:
Do we have to copy this function?

Reply:
Not necessary, this bar is intuitive, this is the only reason. Shall I remove that?